### PR TITLE
Use GeoJson and own types

### PR DESCRIPTION
**Pull request by Beck-berry**
Changes:
* introduced some own types (for popups, trips - which is already on main, too)
* usage of GeoJson types 

After a first review, I'll merge the main and fix the conflicts.

### CODE REVIEW

**Comment by Beck-berry** [comment]
@michaeljohansen There's one thing, I could not solve on my own:
```ts
src/components/BikeRoutePopup.tsx:31:11 - error TS2488: Type '{ [name: string]: any; } & { network?: Network | undefined; from?: string | undefined; to?: string | undefined; description?: string | undefined; website?: string | undefined; }' must have a '[Symbol.iterator]()' method that returns an iterator.
```
Can you suggest a solution?

File: src/components/Map.tsx [hunk 1]
```diff
@@ -40,28 +42,19 @@ const MapContainer = () => {
...
-  // TODO: These four route states are related and can be combined into one
-  //       state object like `const [trip, setTrip] = useState<Trip|null>(null)`
-  const [routeDuration, setRouteDuration] = useState<number | null>(null);
-  const [routeDistance, setRouteDistance] = useState<number | null>(null);
-  const [routeElevation, setRouteElevation] = useState<number | null>(null);
-  const [routeElevationProfile, setRouteElevationProfile] = useState<
-    number[] | null
-  >(null);
+  const [popup, setPopup] = useState<PopupProps | PopupPropsForBikeRoute | null>(null);
+  const [trip, setTrip] = useState<Trip | null>(null);
```
Comment (michaeljohansen) [inline src/components/Map.tsx:57]: Awesome cleanup! 👏

File: src/components/Map.tsx [hunk 2]
```diff
@@ -174,54 +167,48 @@ const MapContainer = () => {
...
-    roadSnow: SimulationFeature[]
+    roadOk: SnowPlowFeature[],
+    roadWarn: SnowPlowFeature[],
+    roadSnow: SnowPlowFeature[]
   ) => {
     if (map.current !== null) {
-      // @ts-expect-error TODO: Are we sure setData works here?
-      map.current.getSource("snow-plow-ok")?.setData({
+      const okRoadSource = map.current.getSource("snow-plow-ok") as GeoJSONSource;
+      okRoadSource.setData({
```
Comment (michaeljohansen) [inline src/components/Map.tsx:201]: Great that you resolved these TODO's!

File: src/components/Map.tsx [hunk 3]
```diff
@@ -338,67 +323,81 @@ const MapContainer = () => {
...
-        const feature = sykkelHotelFeatures[0].properties;
-        setPopupType(SYKKELHOTEL_POPUP);
-        setPopupCoords(event.lngLat);
-        setPopupPoint(feature);
+        setPopup({
+          type: SYKKELHOTEL_POPUP,
+          onClose: onPopupClose,
+          lngLat: [event.lngLat.lng, event.lngLat.lat],
+          point: sykkelHotelFeatures[0].properties
+        });
```
Comment (michaeljohansen) [inline src/components/Map.tsx:338]: Looks much better with a single `setPopup()`.

**Comment by michaeljohansen** [comment]
This all looks great. Nice job! 😄  As long as `npm run typecheck` is happy then go ahead and hit the big green button. ✅

**Review by michaeljohansen — APPROVED** [review]
This all looks great. Nice job! 😄 As long as `npm run typecheck` is happy (and merge goes fine) then go ahead and hit the big green button. ✅

**Pull request closed without merging**